internal/slicer/slicer.go

```diff
@@ -82,19 +82,13 @@ func Run(options *RunOptions) (*Report, error) {
 
 	// Build information to process the selection.
 	extract := make(map[string]map[string][]deb.ExtractInfo)
-	archives := make(map[string]archive.Archive)
+	archives, err := selectPkgArchives(options.Archives, options.Selection)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Can you move this above `extract` and the comment? I think this is a distinct state of selecting the archives not "building information to process the selection", and `extract ` being above might lead you to think it is used when it is not.